src/ngraph/util.cpp

```diff
@@ -268,25 +250,64 @@ ngraph::FpropCache ngraph::cache_fprop(std::shared_ptr<ngraph::Function> fprop,
             throw ngraph_error("Expected op::Result values for op::Result keys in node_param_map");
         }
         cloned_results.push_back(result);
+        result_nodes.push_back(result);
     }
 
-    // get clone bprop parameters
-    op::ParameterVector bprop_input_params;
-    for (auto param : bprop_inputs)
+    auto get_bprop_params = [&bprop_inputs, &fprop_cache]() {
```

Comment:
Someday I want to move the fprop cache out of util.?pp into something like fprop_cache.?pp. Not needed now, just someday...